app/assets/javascripts/snowcrash/pages/nodes/new_form.js

```diff
@@ -0,0 +1,38 @@
+(function() {
+  var copyOver = function($to, from, $typeTo, $typeFrom) {
+    if ($to.val() === '' && from !== '') {
+      $to.val(from.split('\n')[0] + '\n');
+      $typeTo.val($typeFrom.val());
+
+      // The click function we're in actually takes focus when clicked. If
+      // we called focus within the function it gets cancelled out. We give
+      // it a brief timout to allow it to occur after this function ends
+      // but seemingly at the same time.
+      setTimeout(function() {
```

Comment:
@brianp This hack gets a pass for me for its brevity. However, I think we can use JS promise or Jquery deferred objects for this purpose by ensuring the `focus` call. I personally haven't used those outside of ajax calls, so if it gets too complicated, I say we go with the current approach.

Reply:
I like this idea, but as far as actual Promises, anything that is defined within the `click` jquery function, will still finish within the function so
- the promise will finish
- the focus callback will finish
- then the click will finish and take focus back

Jquery deferred looks like a better candidate, but we require the modal scope, which is found within the click event, again bounding us to the inner scope of the click event which will always be the last scope to finish, and taking focus.

I like the idea overall but yeah, to complicated for the gain.

Reply:
I actually decided to try this, and for no reason I could comprehend once I added the promise, double click events started firing. And due to the situation outlined above, focus was still never called at the appropriate time because `click` always finishes last.